crates/rspack_tracing/src/perfetto.rs

```diff
@@ -0,0 +1,161 @@
+use std::{collections::HashMap, io::Write};
+
+use rspack_tracing_perfetto::{
+  idl,
+  idl::TrackDescriptor,
+  idl_helpers::{
+    create_event, create_track_descriptor, custom_scope_packet, unique_uuid, DebugAnnotations,
+  },
+  prost::Message,
+  BytesMut, PerfettoLayer,
+};
+static JAVASCRIPT_ANALYSIS_TRACK: &str = "JavaScript Analysis";
+use crate::{tracer::Layered, Tracer};
+#[derive(Default)]
+pub struct PerfettoTracer {
+  track_state: HashMap<u32, (TrackDescriptor, idl::Trace)>,
+  event_id_map: HashMap<u32, u64>,
+  writer: Option<std::fs::File>,
```

Comment:
it seems writer can't be None

Reply:
yes, but cause it's lazy initialized in setup call, so we have to set it to None during constructor